base/logging/logging.jl

```diff
@@ -41,13 +41,39 @@ filtering.
 function handle_message end
 
 """
-    shouldlog(logger, level, _module, group, id)
+    shouldlog(logger, level, _module, group, id, maxlog)
 
 Return `true` when `logger` accepts a message at `level`, generated for
 `_module`, `group` and with unique log identifier `id`.
 """
 function shouldlog end
 
+# TODO: All of these functions also need an invoke boundary for normal usage...
+
+"""
+    make_normalization_context(logger, nkwargs::Int, maxlog::Union{Int,Nothing})
+
+Generate a "context" object, to be passed in as the second argument of
+`normalize_message` and `normalize_kwarg`.
+"""
+make_normalization_context(logger::AbstractLogger, nkwargs::Int) = nothing
+
+"""
+    normalize_message(logger, context, message)
+"""
+normalize_message(logger::AbstractLogger, context::Any, message::Any) = message
```

Comment:
```suggestion
normalize_message(logger::AbstractLogger, context::Any, message::Any) = convert(Union{String,Base.AnnotatedString{String}}, message)::Union{String,Base.AnnotatedString{String}}
```

Otherwise aren't you at risk for `StackOverflowError`?

Likewise with `normalize_kwarg` below.